### PR TITLE
EMPRO opt out UI

### PR DESCRIPTION
**Pull request by achen2401**
https://movember.atlassian.net/browse/TN-3252
- Frontend calls the trigger api, i.e. `/api/patient/[userId]/trigger` to retrieve the latest triggers and check if there is any that has `_sequential_hard_trigger_count` > 1 (meaning then that it has been asked previously) (see example triggers JSON):
```
{
 ...
  "timestamp": "2023-07-31T23:17:03+00:00",
  "triggers": {
    "domains": {
           "anxious": {
              "ironman_ss.11": "hard",
              "ironman_ss.12": "hard",
              "ironman_ss.13": "hard",
              "_sequential_hard_trigger_count": 3
            },
           "fatigue": {
              "ironman_ss.10": "hard",
              "ironman_ss.9": "hard",
              "_sequential_hard_trigger_count": 2
            },
          "general_pain": {
                "ironman_ss.1": "hard",
                "ironman_ss.2": "hard",
                "ironman_ss.3": "hard",
                "_sequential_hard_trigger_count": 3
              },
           "joint_pain": {
              "ironman_ss.4": "hard",
              "ironman_ss.5": "hard",
              "ironman_ss.6": "hard",
              "_sequential_hard_trigger_count": 2
            },
          "sad": {
            "ironman_ss.17": "hard",
            "ironman_ss.18": "hard",
            "ironman_ss.19": "hard",
            "_sequential_hard_trigger_count": 3
          },
    }
   }
...
}
```
- If it finds any qualifying domains, it will present the OPT-OUT UI modal to the user, with the checkbox(es) that allow user to select which domain(s) to opt out:
OptoutInputs

- frontend submits the domain(s) that user has chosen to opt out to the backend API, `/api/patient/<user_id>/triggers/opt_out`, to be saved on the database for later use (e.g. use in clinician email, adherence report)

- One user submits the request, the EMPRO thank you modal that follows will display which(s) domains the user has chosen to opt out:
OptoutChosen

### CODE REVIEW

**Comment by achen2401** [comment]
@pbugni this is an example of the data that the frontend will be sending back to the backend, once the user has selected one or more domains to opt out:
```
{
    "user_id": 4105,
    "visit_month": 1,
    "opt_out_domains": [
        "fatigue",
        "general_pain"
    ]
}
```
Let me know if that works, or change is needed?

**Comment by pbugni** [comment]
Hi @achen2401 , yes, let's use ad-hoc JSON such as the following.  This would set it for the same two domains only, from your example above.
  
I'm attempting to use the same format you get from `https://eproms.truenth.org/api/patient/<id>/triggers`, but simplified down to just set the opt-out domains of choice:

```
{
  "triggers": {
    "domain": {
      "fatigue": {
        "_opt_out_next_visit": "true"
      },
      "general_pain": {
        "_opt_out_next_visit": "true"
      },
      "user_id": 1234,
      "visit_month": 7
    }
  }
}
```

please PUT/POST that to `/api/patient/<user_id>/triggers/opt_out`  (which isn't ready yet, but will be soon)

**Comment by achen2401** [comment]
> please PUT/POST that to `/api/patient/<user_id>/triggers/opt_out` (which isn't ready yet, but will be soon)
@pbugni thank you, noting that frontend now calls this API.  I have modified the code accordingly.

File: portal/static/js/src/empro.js [hunk 1]
```diff
@@ -1,204 +1,546 @@
...
+  EmproObj.submittedOptOutDomains = EmproObj.selectedOptOutDomains;
+  EmproObj.populateSelectedOptoutUI();
+  EmproObj.initOptOutModal(false);
+  EmproObj.initThankyouModal(true);
+  return true;
+};
+emproObj.prototype.handleSubmitOptoutData = function () {
+  if (!EmproObj.hasSelectedOptOutDomains()) {
+    EmproObj.setOptoutError(
+      "Please check at least one checkbox. Otherwise click 'Continue'."
```
Comment (mcjustin) [inline portal/static/js/src/empro.js:190]: Is this referring to the button labelled "DISMISS", not "Continue"?

**Review by mcjustin — APPROVED** [review]
Thanks @achen2401  just had one question...

File: portal/static/js/src/empro.js (hunk 1 above)
Comment (achen2401) [reply portal/static/js/src/empro.js]: thank you @mcjustin.  I have fixed the text now so when the user didn't select a checkbox, this is what will show:
Error

Does the "Continue" button seem redundant? If so, I will remove.

Comment (mcjustin) [reply portal/static/js/src/empro.js]: It seems redundant. Thanks for this screenshot... stepping back a bit, I don't recall that they should check at least one checkbox. Do you see that documented anywhere? 
If it's not a requirement, then if they click DISMISS and they haven't selected anything yet, perhaps we should show text that says "Are you certain that you don't want to select anything? If so, click DISMISS again."

Comment (achen2401) [reply portal/static/js/src/empro.js]: @mcjustin right, I don't recall either, that it is a requirement.  I made the changes and attached is the latest screenshot after the user clicks DISMISS and hasn't selected anything:
dismiss

Comment (mcjustin) [reply portal/static/js/src/empro.js]: Looks good to me, thanks Amy!

**Review by pbugni — APPROVED** [review]
thanks @achen2401

**Pull request merged**